packages/migrate/migrate.ts

```diff
@@ -0,0 +1,112 @@
+import extractTextPluginTransform from "./extractTextPlugin/extractTextPlugin";
+import noEmitOnErrorsPluginTransform from "./noEmitOnErrorsPlugin/noEmitOnErrorsPlugin";
+import removeDeprecatedPluginsTransform from "./removeDeprecatedPlugins/removeDeprecatedPlugins";
+
+interface ItransformsObject {
+	loadersTransform: object;
+	resolveTransform: object;
+	removeJsonLoaderTransform: object;
+	uglifyJsPluginTransform: object;
+	loaderOptionsPluginTransform: object;
+	bannerPluginTransform: object;
+	extractTextPluginTransform: object;
+	noEmitOnErrorsPluginTransform: object;
+	removeDeprecatedPluginsTransform: object;
+}
+
+const transformsObject: ItransformsObject = {
+	loadersTransform,
+	resolveTransform,
+	removeJsonLoaderTransform,
+	uglifyJsPluginTransform,
+	loaderOptionsPluginTransform,
+	bannerPluginTransform,
+	extractTextPluginTransform,
+	noEmitOnErrorsPluginTransform,
+	removeDeprecatedPluginsTransform
+};
+
+export const transformations = Object.keys(transformsObject).reduce((res: object, key: string) => {
+	res[key] = (ast: object, source: string) =>
+		transformSingleAST(ast, source, transformsObject[key]);
+	return res;
+}, {});
+
+/**
+ *
+ * Transforms single AST based on given transform function
+ * and returns back a promise with resolved transformation
+ *
+ * @param {Object} ast - AST object
+ * @param {String} source - source string
+ * @param {Function} transformFunction - Transformation function with source object
+ * @returns {Object} PLazy promise with resolved transform function
+ */
+
+export const transformSingleAST = (
+	ast: object,
+	source: string,
+	transformFunction: (jscodeshift: object, ast: object, source: string) => object)
+	: PLazy<{}> => {
+
+	return new PLazy((
+		resolve: (value?: {} | PromiseLike<{}>) => void,
+		reject: (reason?: object) => void
+	) => {
+		setTimeout(_ => {
+			try {
+				resolve(transformFunction(jscodeshift, ast, source));
+			} catch (err) {
+				reject(err);
+			}
+		}, 0);
+	});
+}
+
+/**
+ *
+ * Transforms a given piece of source code by applying selected transformations to the AST.
+ * By default, transforms a webpack version 1 configuration file into a webpack version 2
+ * configuration file.
+ *
+ * @param {String} source - source file contents
+ * @param {Function[]} [transforms] - List of transformation functions, defined in the
+ * order to apply them in. By default, all defined transformations.
+ * @param {Object} [options] - recast formatting options
+ * @returns {Promise} promise functions for series
+ */
+
+export const transform = (
+	source: string,
+	transforms: Iterable<Function | PromiseLike<Function>>,
+	options: object)
+	: Promise<Function[]> => {
+
+	const ast: any = jscodeshift(source);
+	const recastOptions: object = Object.assign(
+		{
+			quote: "single"
+		},
+		options
+	);
+	transforms =
+		transforms || Object.keys(transformations).map(k => transformations[k]);
+
+	return pEachSeries(transforms, (f: Function) => f(ast, source))
+		.then(_ => {
```

Comment:
missing?

Reply:
That was the pattern followed so far where if we didn't use anything from the resolved promise, it was decided to use `_ =>` instead of `() =>`.
Now that there's typescript, I am more against this pattern. What's `_`? What's his meaning? We put `any` just because we want to leave `_`? I think we should drop it

Reply:
I also think `() =>` is better. Should I replicate this in all occurrences?

Reply:
Let's see how @ev1stensberg feels about it

Reply:
`void` is when your function returns nothing. I am not sure if it's possible to assign it to a parameter... never seen it. Maybe typescript supports it

Reply:
If void is supported, keep underscore, if not, remove it